common/projectconfig/integration_test.go

```diff
@@ -59,3 +59,37 @@ func TestFindConfig(t *testing.T) {
 		checkConfig("."),
 	)
 }
+
+func TestConfigValidation(t *testing.T) {
+	in.Run(t, "./validateconfig/ftl-project.toml",
+		in.CopyModule("validateconfig"),
+
+		// Global sets never error.
+		in.Chdir("validateconfig", in.Exec("ftl", "config", "set", "key", "--inline", "valueTwo")),
+		in.Chdir("validateconfig", in.Exec("ftl", "config", "set", "key", "--inline", "2")),
+		in.ExecWithExpectedOutput("\"2\"\n", "ftl", "config", "get", "key"),
```

Comment:
nit: backticks avoid the need to escape the double quotes inside the string (here and below) 
```
`"string"\n`
```

Reply:
can't use escape characters with backticks unfortunately, so that `\n` would be two chars

Reply:
ahhhh drats